apitest/scripts/trade-simulation.sh

```diff
@@ -0,0 +1,228 @@
+#
+#  - Bisq must be fully built with apitest dao setup files installed.
+#    Build command:  `./gradlew clean build :apitest:installDaoSetup`
+#
+#  - All supporting nodes must be run locally, in dev/dao/regtest mode:
+#           bitcoind, seednode, arbdaemon, alicedaemon, bobdaemon
+#
+#    These should be run using the apitest harness.  From the root project dir, run:
+#    `$ ./bisq-apitest --apiPassword=xyz --supportingApps=bitcoind,seednode,arbdaemon,alicedaemon,bobdaemon --shutdownAfterTests=false`
+#
+#  - Only regtest btc can be bought or sold with the test payment account.
+#
+# Usage:
+#
+#  This script must be run from the root of the project, e.g.:
+#
+#     `$ apitest/scripts/trade-simulation.sh -d buy -c fr -m 3.00 -a 0.125`
+#
+#  Script options:  -d <direction> -c <country-code> -m <mkt-price-margin(%)> - f <fixed-price> -a <amount(btc)>
+#
+# Examples:
+#
+#    Create a buy/eur offer to buy 0.125 btc at a mkt-price-margin of 0%, using an Italy face to face payment account:
+#
+#       `$ apitest/scripts/trade-simulation.sh -d buy -c it -m 0.00 -a 0.125`
+#
+#    Create a sell/eur offer to sell 0.125 btc at a fixed-price of 38,000 euros, using a France face to face
+#    payment account:
+#
+#       `$ apitest/scripts/trade-simulation.sh -d sell -c fr -f 38000 -a 0.125`
+
+export APP_BASE_NAME=$(basename "$0")
+export APP_HOME=$(pwd -P)
+export APITEST_SCRIPTS_HOME="${APP_HOME}/apitest/scripts"
+
+# Source the env and some helper functions.
+. "${APITEST_SCRIPTS_HOME}/trade-simulation-env.sh"
+. "${APITEST_SCRIPTS_HOME}/trade-simulation-utils.sh"
+
+checksetup
+parseopts "$@"
+
+printdate "Started ${APP_BASE_NAME} with parameters:"
+printscriptparams
+printbreak
+
+registerdisputeagents
+
+printdate "Alice looks for the ID of the face to face payment account method (Bob will use same payment method)."
+CMD="${CLI_BASE} --port=${ALICE_PORT} getpaymentmethods"
+printdate "ALICE CLI: ${CMD}"
+getpaymentaccountmethods "$CMD"
+printbreak
+
+printdate "Alice uses the F2F payment method id to create a face to face payment account in country ${COUNTRY_CODE}."
+CMD="${CLI_BASE} --port=${ALICE_PORT} getpaymentacctform --payment-method-id=F2F"
+printdate "ALICE CLI: ${CMD}"
+getpaymentaccountform "$CMD"
+printbreak
+
+printdate "Bob & Alice edit their ${COUNTRY_CODE} payment account forms, and renames them to ${F2F_ACCT_FORM}"
+editpaymentaccountform "$COUNTRY_CODE"
+cat "${APITEST_SCRIPTS_HOME}/${F2F_ACCT_FORM}"
+
+# Remove the autogenerated json template because we are going to use one created by a python script in the next step.
+CMD="rm -v ${APP_HOME}/f2f_*.json"
+DELETE_JSON_TEMPLATE=$(${CMD})
+echo "$DELETE_JSON_TEMPLATE"
+printbreak
+
+printdate "Bob and Alice create their face to face ${COUNTRY_CODE} payment accounts."
+CMD="${CLI_BASE} --port=${BOB_PORT} createpaymentacct --payment-account-form=${APITEST_SCRIPTS_HOME}/${F2F_ACCT_FORM}"
+printdate "BOB CLI: ${CMD}"
+CMD_OUTPUT=$(createpaymentacct "${CMD}")
```

Comment:
These references are a bit more complicated to quote because of nested variable references. Escaping the inner quotes should do the trick: `CMD_OUTPUT="$(createpaymentacct \"${CMD}\")"`. Readability isn't great. Alternatively, at the risk of too much vertical verbosity, you could split the inner part into a new variable: `tmp="createpaymentacct $CMD"`.

Reply:
Will you tell me why I need to make this change?  It is working.  
What is wrong with  `CMD_OUTPUT=$(createpaymentacct "${CMD}")`?

Reply:
It's the same variable referecing quoting issue I refer to in other comments. It prevents reinterpretation of some special characters, prevents word splitting. See examples here: https://tldp.org/LDP/abs/html/quotingvar.html